### ISSUE TITLE
fail to create ns secret

### ISSUE THREAD

**Issue by adrienb4** [nsupdate-info/nsupdate.info#453]
Hi

We can't generate a ns secret. 

![image](https://user-images.githubusercontent.com/15191572/72623090-c7bf8c00-3944-11ea-850f-59e50d78e84c.png)

Logs :
```
django.db.utils.DataError: (1406, "Data too long for column 'nameserver_update_secret' at row 1")
```

HMAC_SHA512 secret need to have a length to 89, but you set it to 88 in the last migration file.

![image](https://user-images.githubusercontent.com/15191572/72623577-ab701f00-3945-11ea-96ce-a2aca0a48ba6.png)

Commit #0a625d685358fa668a46eea63b1c73ef84a6442b

Thx

**Comment by ThomasWaldmann** [comment]
@elnappo ^^^

**Comment by elnappo** [comment]
Hmm, we haven't changed the length of this field. It was always 88
https://github.com/nsupdate-info/nsupdate.info/blob/master/src/nsupdate/main/models.py#L103
https://github.com/nsupdate-info/nsupdate.info/search?q=nameserver_update_secret&unscoped_q=nameserver_update_secret

**Comment by adrienb4** [comment]
In my case all HMAC-SHA512 secrets have 89 for length.
I recently upgraded nsupdate and now I can't add new domains.
All domains secret are been generated by nsupdate before upgrading.

**Comment by elnappo** [comment]
On nsupdate.info my nameserver update secrets are 88 chars long. @ThomasWaldmann do you have an idea?

**Comment by ThomasWaldmann** [comment]
I tried to add a new domain: len = 88

Then I created a new configuration / new secret for the same domain: len = 88, too.

So, I can't reproduce.

**Comment by ThomasWaldmann** [comment]
@adrienb4 how do your generated secrets look like?

**Comment by ThomasWaldmann** [comment]
is it related to #447?

**Comment by adrienb4** [comment]
For example with a domain secret created by nsupdate : 
```
SmtQUHpzWW5mZlZRNkdnYmJabnBFQ01nc21hN3VrekZGTWR0RVU2RjVZOGtXdlJaNDR4SkIzQkZIRVE0SDZGTQ==
```
Len is 89.

![image](https://user-images.githubusercontent.com/15191572/72784782-dd290480-3c29-11ea-9969-d624f789a269.png)

**Comment by adrienb4** [comment]
I just generated a new secret domain in CLI : 
``` 
dnssec-keygen -a HMAC-SHA512 -b 512 -n HOST tata.dyn.toto.fr
```

Len is 89 too

```
uP4TqSMiUQUkPrInsaI77WH6up28yKI+36Kp44Aq3e9P+JDQ4toXpCRxCTkU8lhSJyLL6lGbXgoX6WWNjOB/AQ==
```

**Comment by adrienb4** [comment]
Hummm as much for me :/

```
echo -ne tata.dyn.toto.fr | openssl dgst -sha512 -hmac "key" -binary | base64 -w0 | wc -c
88
```
Vscode misled me in error. The first character is 1 not 0.

But why I can't generate a new secret ? I have no problem if I set it in the Django admin interface.

**Comment by adrienb4** [comment]
Hummm as much for me :/
```
echo -n 'uP4TqSMiUQUkPrInsaI77WH6up28yKI+36Kp44Aq3e9P+JDQ4toXpCRxCTkU8lhSJyLL6lGbXgoX6WWNjOB/AQ==' | wc -c
88
```

```
echo 'uP4TqSMiUQUkPrInsaI77WH6up28yKI+36Kp44Aq3e9P+JDQ4toXpCRxCTkU8lhSJyLL6lGbXgoX6WWNjOB/AQ==' | wc -c
89
```
Vscode misled me in error. The first character is 1 not 0.

**Comment by adrienb4** [comment]
We have found the origin of the problem.

We think when you have upgrade Django to 2.2.9, you introduce a bug with domain secret.
Django tries to update domain secret with a wrong format. The expected format is a string, however we have a binary.

Domain secret in str mode has a length of 88. In binary the length is 91.

![image](https://user-images.githubusercontent.com/15191572/72809059-29406d00-3c5b-11ea-9036-062cb294f127.png)


I also discovered that sqlite was not looking at the size of these fields.... You can put more than 88 in this case. Not possible with MySQL (my backend).

![image](https://user-images.githubusercontent.com/15191572/72808698-62c4a880-3c5a-11ea-9670-58d872cc71f1.png)

See : https://sqlite.org/faq.html#q9

**Comment by ThomasWaldmann** [comment]
ok, guess it is fixed by #454 - but do we now have invalid secret data in the (sqlite) db?
and why did that even work when reading b'...' from the db?

**Comment by brice-gros** [comment]
good point, I assume that switching from django 1 to 2, make the secrets in sqlite invalid anyway, invalidating any comparison string vs bytes.
By the way, regarding #445, switching from sqlite to postgresql (or mysql), won't be possible neither, due to the insertion failing since the varchar's max length constraint is not respected.
So, a migration step verifying and fixing nameserver_update_secret would fix the whole issue, I guess

**Comment by ThomasWaldmann** [comment]
I suspect that even if we inserted b'...' at some point in time, that must have been fixed somehow. At least I did not find any such entries via django admin on www.nsupdate.info.

Guess users/admins who added such entries noticed that they are not working and manually removed the `b''`.
